libs/components/core/src/lib/modules/sr-label/sr-label.directive.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  DestroyRef,
+  Directive,
+  ElementRef,
+  Input,
+  OnInit,
+  inject,
+} from '@angular/core';
+
+const SR_LABELS_CONTAINER_ID = 'sky-sr-labels-container';
+
+/**
+ * Adds the element to a screen reader only section of the body.
+ * This prevents components' DOM from including text only intended for screen readers.
+ */
+@Directive({
+  selector: '[skySrLabel]',
+  standalone: true,
+})
+export class SkySrLabelDirective implements OnInit {
+  readonly #elementRef = inject(ElementRef);
+  #_createLabel = true;
+
+  public ngOnInit(): void {
+    this.#insertSrLabel();
+  }
+
+  @Input()
+  public set destroyRef(ref: DestroyRef) {
+    ref.onDestroy(() => this.#removeSrLabel());
+  }
+
+  @Input()
+  public set createLabel(value: boolean | undefined) {
```

Comment:
I had to go this `createLabel` route instead of putting an `*ngIf` on the label element because this will likely always be used in conjunction with `skyId`, and a template cannot reference an element that has any form of structural directive on it, because it interprets that as meaning the element might not be in the DOM to reference--even when we know it is there. You cannot `?.` or assert it is there or anything. It just won't let you reference it.